### PR TITLE
Imap backend

### PR DESCRIPTION
**Pull request by kha-white**

### CODE REVIEW

File: Core/Models/ImapClientModel.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            _port = port;
+            _useSsl = useSsl;
+            _connected = false;
+            _messages = new Dictionary<Tuple<string, UniqueId>, MimeKit.MimeMessage>();
+        }
+
+        public void Connect()
+        {
+            _client = new ImapClient();
+            _client.ServerCertificateValidationCallback = (s, c, h, e) => true;
```
Comment (Pawcv) [inline Core/Models/ImapClientModel.cs]: Is it safe to ignore SSL certificate?

File: Core/Models/ImapClientModel.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            _client.Connect(_host, _port, _useSsl);
```
Comment (Pawcv) [inline Core/Models/ImapClientModel.cs]: If Connect raise exception _client should be disposed, and exception passed up

**Review by Pawcv — CHANGES_REQUESTED** [review]
Some changes needs to be done in Connect method, ImapClient should be disposed safely if exception is raised, maybe ImapClient can be created in constructor and ImapClientModel will implement IDisposable

File: Core/Models/ImapClientModel.cs (hunk 1 above)
Comment (kha-white) [reply Core/Models/ImapClientModel.cs]: I would leave it for now and implement it later

**Review by Pawcv — APPROVED** [review]

**Pull request merged**